### ISSUE TITLE
MultiSelectComboBox with material design

### ISSUE THREAD

**Issue by chesh111re** [RWS/Multiselect-ComboBox#109]
Hi!
If you create a new wpf project and add material design and multiselectcombobox via nuget. Then add this code to App.xaml

```
<Application x:Class="WpfApp1.App"
             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:local="clr-namespace:WpfApp1"
             xmlns:materialDesign="http://materialdesigninxaml.net/winfx/xaml/themes"
             StartupUri="MainWindow.xaml">
    <Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <materialDesign:BundledTheme BaseTheme="Light" PrimaryColor="Green" SecondaryColor="Lime" />

                <ResourceDictionary Source="pack://application:,,,/MaterialDesignThemes.Wpf;component/Themes/MaterialDesign2.Defaults.xaml" />
                <ResourceDictionary Source="/ReactiveValidation.Wpf;component/Themes/Generic.xaml" />
                <ResourceDictionary Source="/Sdl.MultiSelectComboBox;component/Themes/Generic/MultiSelectComboBox.xaml"></ResourceDictionary>
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>
    </Application.Resources>
</Application>
```

In MainWindow.xaml add MultiSelectComboBox, you get strange behavior.

```
<Grid>
    <sdl:MultiSelectComboBox ItemsSource="{Binding Colors}"
                             SelectedItems="{Binding SelectedColors}"
                             EnableAutoComplete="False"
                             SelectionMode="Multiple" />
</Grid>
```

![vMqmghZ6IU](https://github.com/user-attachments/assets/92c4ef89-b50a-4028-a930-b01b5b333abe)

I think the problem is in the styles: toggle button width or textblock width. Have you ever encountered such behavior?

**Comment by sdolhaor** [comment]
I confirm the issue, and attempted a fix. (By default the MaterialDesign theme was apparently injecting a fixed width for ToggleButton instances, where we wanted an Auto width.)

**Comment by PeskovV** [comment]
Thank you a lot!

**Comment by PeskovV** [comment]
@sdolhaor can you please tell me when the release will be in nuget?

**Comment by sdolhaor** [comment]
This should have been automatic, but apparently there is an issue with the pipeline. It was now run manually, the new nuget is published now, version 1.0.101. Apologizes for the delay.

**Comment by mitchcapper** [comment]
@sdolhaor while this fixed material I found going down the path of manually setting the properties was probably more work than it is worth.  With .net having native fluent theme support in .NET 9+  or other themes like WPF-UI its likely a losing battle to try and manually set any property they may change.   Given the fact things like togglebutton/textbox used in the control are not really used in the traditional UI sense forcing them to the WPF default style prevents any properties from getting changed.   See my notes in #112 at the end.  I think, if desired, you could use dynamic to allow the style to be overwritten by the user somewhat easily ie:
```
<Style x:Key="MultiSelectComboBox.FilterTextBoxBaseStyle"
       TargetType="{x:Type TextBox}"
       BasedOn="{StaticResource {x:Type TextBox}}">
    <Setter Property="Margin" Value="0" />
</Style>

<Style BasedOn="{DynamicResource MultiSelectComboBox.FilterTextBoxBaseStyle}"
       TargetType="{x:Type TextBox}">
    <Setter Property="Margin" Value="0" />
</Style>
```
Then the user could change MultiSelectComboBox.FilterTextBoxBaseStyle to their own liking.
